snuba/pipeline/simple_pipeline.py

```diff
@@ -75,6 +75,7 @@ def __init__(self, query_plan_builder: ClickhouseQueryPlanBuilder) -> None:
     def build_execution_pipeline(
         self, request: Request, runner: QueryRunner
     ) -> QueryExecutionPipeline:
+        assert isinstance(request.query, LogicalQuery)
```

Comment:
Seems we should be able to remove the dependency on Request and do like EntityQueryPlanner which takes query and settings instead of depending on request. Then the type problem does not exist anymore.

Reply:
It creates a new type problem, which is that the `build_execution_pipeline` has to take a `Union[LogicalQuery, CompositeQuery]` regardless of whether it's a simple or complex pipeline. Not sure if that's better or not.

Reply:
> build_execution_pipeline has to take a Union[LogicalQuery, CompositeQuery]

Which is what it should doo but the delegate one does not support it so nevermind. This will need a different fix. Sorry about this.